main.go

```diff
@@ -0,0 +1,453 @@
+	var keyBytesPool [][]byte
+	if keysFiles != "" {
+		for _, k := range strings.Split(keysFiles, ",") {
+			svcAccountJSONBytes, err := ioutil.ReadFile(k)
+			if err != nil {
+				return nil, nil, err
+			}
+			keyBytesPool = append(keyBytesPool, svcAccountJSONBytes)
+		}
+	}
+
+	if impersonatedAccounts != "" {
+		serviceAccountPool = strings.Split(impersonatedAccounts, ",")
+	}
+
+	return serviceAccountPool, keyBytesPool, nil
+}
+
+func getRandomServiceAccount(accounts []string, keys [][]byte) (string, []byte, error) {
+	var selectedAccount string
+	var selectedKey []byte
+	if len(accounts) > 0 {
+		selectedAccount = accounts[rand.Intn(len(accounts))]
+		glog.V(50).Infof("             Selecting serviceAccount: %s", selectedAccount)
+	}
+	if len(keys) > 0 {
+		keyIndex := rand.Intn(len(keys))
+		selectedKey = keys[keyIndex]
+		glog.V(50).Infof("             Selecting serviceAccount keyIndex value: %d", keyIndex)
+	}
+	// this should not happen
+	if selectedAccount == "" && selectedKey == nil {
+		return "", nil, fmt.Errorf("no valid service account found")
+	}
+	return selectedAccount, selectedKey, nil
+}
+
+func findDomainUsers(ctx context.Context, cx string, searchFilter string, adminService *admin.Service) ([]*admin.User, error) {
+
+	allUsers := make([]*admin.User, 0)
+	pageToken := ""
+	q := adminService.Users.List().Customer(cx).Query(searchFilter)
+	for {
+		if pageToken != "" {
+			q = q.PageToken(pageToken)
+		}
+		r, err := q.Do()
+		if err != nil {
+			return nil, err
+		}
+		for _, u := range r.Users {
+			glog.V(20).Infof("      Found User: %s", u.PrimaryEmail)
+			allUsers = append(allUsers, u)
+		}
+		pageToken = r.NextPageToken
+		if pageToken == "" {
+			break
+		}
+	}
+	return allUsers, nil
+}
+
+func findResourcesByAssetType(ctx context.Context, organizationID string, assetType string, query string, assetClient *asset.Client) (map[string]*assetpb.ResourceSearchResult, error) {
+
+	resourceList := make(map[string]*assetpb.ResourceSearchResult)
+
+	req := &assetpb.SearchAllResourcesRequest{
+		Scope:      fmt.Sprintf("organizations/%s", organizationID),
+		Query:      query,
+		AssetTypes: []string{assetType},
+	}
+
+	it := assetClient.SearchAllResources(ctx, req)
+	for {
+		response, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		switch {
+		case assetType == assetTypeOrganization:
+			orgName := strings.TrimPrefix(response.Name, "//cloudresourcemanager.googleapis.com/organizations/")
+			glog.V(20).Infof("     Found Organization %s", orgName)
+			resourceList[fmt.Sprintf("organizations/%s", orgName)] = response
+		case assetType == assetTypeProject:
+			projectID := strings.TrimPrefix(response.Name, "//cloudresourcemanager.googleapis.com/projects/")
+			glog.V(20).Infof("     Found projectID %s", projectID)
+			resourceList[projectID] = response
+		default:
+			return nil, fmt.Errorf(fmt.Sprintf("Error getting resources:  unknown assetType: %s", assetType))
```

Comment:
No need for `fmt.Errorf(fmt.Sprintf(...))`, you can just do `fmt.Errorf(...)` directly.

```suggestion
			return nil, fmt.Errorf("error getting resources:  unknown assetType: %s", assetType)
```

Also, https://github.com/golang/go/wiki/CodeReviewComments#error-strings for the capital letter at the start.